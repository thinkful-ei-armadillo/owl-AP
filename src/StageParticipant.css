.stage-participant {
    flex: 1 0 25%;
    margin: 5px;
    height: 100px;
}

.stage-participant-self {
    position: absolute;
    bottom: 0;
    width: 135px; 
    min-height: 90px; 
    max-width: 100%; 
    margin-bottom: 15px; 
    overflow: hidden; 
    position: absolute; 
    left: 50%; 
    bottom: 0; 
    border-radius: 3px; 
    height: 100px;
    background-color: black;
}

.stage-participant-self img {
    height: 50px;
    width: 50px;
}