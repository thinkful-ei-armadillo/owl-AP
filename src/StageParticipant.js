import React from 'react';
import './StageParticipant.css'

export default function StageParticipant(props){
    let classes = (props.self ? "stage-participant-self" : "stage-participant");
 
    return (
        <div className={classes}>
            <div className="info-status">
                {props.name}
            </div>
            <img src={props.avatar} alt={props.name}/>
        </div>
    );
}