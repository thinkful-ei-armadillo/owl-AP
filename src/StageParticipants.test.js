import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import StageParticipant from './StageParticipant';


describe('Participant component testing', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');

    //render the component, this is the actual test, if something is wrong it will fail here
    ReactDOM.render(
      <StageParticipant
      key={1} 
      name="Ronnie" 
      avatar="https://robohash.org/itaquevoluptatumrerum.jpg?size=200x200&set=set1'"
      self={false}
      />,
      div
    );

    //clean up code
    ReactDOM.unmountComponentAtNode(div);
  });

  it('renders the UI as expected', () => {
    const tree = renderer
      .create(
        <StageParticipant
        key={1} 
        name="Ronnie" 
        avatar="https://robohash.org/itaquevoluptatumrerum.jpg?size=200x200&set=set1'"
        self={false}
        />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
