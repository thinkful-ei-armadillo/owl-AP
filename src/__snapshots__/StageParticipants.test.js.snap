// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Participant component testing renders the UI as expected 1`] = `
<div
  className="stage-participant"
>
  <div
    className="info-status"
  >
    Ronnie
  </div>
  <img
    alt="Ronnie"
    src="https://robohash.org/itaquevoluptatumrerum.jpg?size=200x200&set=set1'"
  />
</div>
`;
